crates/wit-component/src/decoding.rs

```diff
@@ -1423,9 +1393,9 @@ impl Registrar<'_> {
         // situation the code path we're currently on will be hit because
         // there's a preexisting definition of `foo` from the import and it's
         // now going to be unified with what we've seen in the export. When
-        // visiting the `list<string>` field of the `foo` union this ends up
-        // being different than the `list<string>` used by the `bar` union. The
-        // reason for this is that when visiting `var` the wasm-defined `(list
+        // visiting the `list<string>` case of the `foo` variant this ends up
+        // being different than the `list<string>` used by the `bar` variant. The
+        // reason for this is that when visiting `bar` the wasm-defined `(list
```

Comment:
I _think_ this preserves the intent of the comment, but I'll admin I didn't try to comprestand the whole thing very well.